UPGRADE.md

```diff
@@ -1411,6 +1431,124 @@ const Layout = (props) => {
+
+The new solution leverage the fact that we already have the `save` function available through context (`useSaveContext`). You can get the current form values using [final-form hooks](https://final-form.org/docs/react-final-form/api/useFormState), call the `save` function directly and override any of the side effects as you usually do with mutation hooks.
+
+### `<SaveButton>` Props and Behavior Have Changed
+
+The `<SaveButton>` already accepted the `onSuccess`, `onFailure` and `transform` props to handle multiple submit buttons. We removed the `handleSubmit` and `handleSubmitWithRedirect` props completely. Instead, when provided any of the side effects props, the `<SaveButton>` will render a simple button and will call the `save` function with them. It also take care of preventing the default form submit.
+
+If you relied on `handleSubmit` or `handleSubmitWithRedirect`, you can now use the `SaveButton` and override any of the side effect props: `onSuccess`, `onFailure` or `transform`.
+
+### The `save` Function Signature Changed
+
+The `save` function signature no longer take a redirection side effect as the second argument. Instead, it only receive the data and an object for side effects (which was the third argument before):
+
+```diff
+const MyCustomCreate = () => {
+    const createControllerProps = useCreateController();
+    const notify = useNotify();
++    const redirect = useRedirect();
+
+    const handleSubmit = (values) => {
+-        createControllerProps.save(values, 'show', {
++        createControllerProps.save(values, {
+            onSuccess: (data) => {
+                notify('Success');
++                redirect('show', '/posts', data.id);
+            }
+        })
+    }
+
+    return (
+        <CreateContextProvider value={createControllerProps}>
+            <Form
+                onSubmit={handleSubmit}
+                render={formProps => (
+                    <form onSubmit={props.handleSubmit}>
+                        ...
+                    </form>
+                )}
+            />
+        </CreateContextProvider>
+    )
+}
+```
+
+### `<FormContext>`, `<FormContextProvider>` and `useFormContext` Have Been Removed
+
+As the `save` and `saving` properties were already available through the `<SaveContext>` component and its `useSaveContext` hook, we removed the `<FormContext>`, `<FormContextProvider>` components as well the `useFormContext` hook.
+
+The functions around form groups management we extracted into the `<FormGroupsProvider>` component and `useFormGroups` hook.
+
+### The `redirect` prop Has Been Removed From `FormWithRedirect`, `SimpleForm`, `TabbedForm` and `SaveButton`
+
+The `FormWithRedirect`, `SimpleForm`, `TabbedForm` and `SaveButton` don't have a `redirect` prop anymore. You should handle redirection either:
+- in the submit handler if you provided a custom one, leveraging the `useRedirect` hook
```

Comment:
As you explicited 3 of the 3 list items with examples below, why don't you do the same fpor the 3rd list item, and femove the list altogether? It'sll be easier to follow. 

Otherwise, for each list item, I think 'how am I going to do this?'